### ISSUE TITLE
bug: status of cluster-scoped workload object is not propagated back

### ISSUE THREAD

**Issue by waltforme** [kubestellar/kubestellar#2413]
### Describe the bug

As shown in the title.

### Steps To Reproduce

Looking at the log of the controller manager, we can see something similar to
```
I0828 03:15:24.209486  202005 workstatus.go:101] "Could not find object" logger="Status.worker-0" objectIdentifier={"GVK":{"Group":"","Version":"v1","Kind":"Namespace"},"Resource":"namespaces","ObjectName":{"Namespace":"","Name":"nginx"}}
```

But namespace objects may _not_ be good representatives of cluster-scoped workload objects, because we may _not_ want the status of a namespace object to be propagated back from WEC to WDS.

### Expected Behavior

Status of cluster-scoped workload object should be able to propagated back from WEC to WDS, if such propagation is said to be expected (or, not prohibited) for that workload object.

### Want to contribute?

- [ ] I would like to work on this issue.

### Additional Context

The cause of the bug is documented [here](https://github.com/kubestellar/kubestellar/pull/2409#discussion_r1723979753). But merely adding a lister for cluster object is not sufficient. In addition to adding the missing lister, Kubestellar should be able to be told that, for each workload object, the status should be propagated or not. I think this is related to #1851 .

**Comment by vMaroon** [comment]
@waltforme namespaces are not propagated on the ocm-status-addon level.

https://github.com/kubestellar/ocm-status-addon/blob/v0.2.0-rc10/pkg/agent/agent.go#L39-L41

The code does support it, but I never really tested it. You may want to test with a different cluster-scoped object.

**Comment by waltforme** [comment]
@vMaroon Exactly, so in the opening comment I wrote that namespaces are not good representatives of cluster-scoped workload objects. I still used namespaces only because everyone can easily reproduce the "Could not find object" message.

On the topic of propagation of Namespace status, I saw @MikeSpreitzer has some different opinion. If I understand correctly, looks like @MikeSpreitzer thinks
- the status should be propagated from WEC to ITS;
- but overwrites should not happen in WDS.

**Comment by MikeSpreitzer** [comment]
Right. I do not understand why Namespace, Service, and ServiceAccount are excepted from propagating status from WEC to WorkStatus. Note also that the code currently does NOT except CustomResourceDefinition.

**Comment by vMaroon** [comment]
@MikeSpreitzer based on the comment above the linked code, I guess whoever wrote it wanted to save the creation of "empty" WorkStatuses.

I see two options here:
1. WorkStatuses are generated for such resources - ocm status addon's responsibility 
2. The status controller explicitly reflects all expected CombinedStatuses, and all entries in them. Missing WorkStatuses/fields would resolve to null - status controller's responsibility 

It feels like the first is a better placement of responsibility, but the second is more robust.

**Comment by MikeSpreitzer** [comment]
@vMaroon: currently the status add-on propagates status back without regard to KubeStellar control objects. I think that the simplest fix is to remove the bogus exception for Namespace, Service, and ServiceAccount.

**Comment by waltforme** [comment]
Resolved by #2414, kubestellar/ocm-status-addon#80, #2417, and #2424. Closing.
